### ISSUE TITLE
Auth Integration for gating specific pages

### ISSUE THREAD

**Issue by studentofcoding** [shuding/nextra#1121]
I recently have a need to Integrate Auth 3rd party like supabase auth, into gating specific pages, Is it already possible to do it right now? Thanks

**Comment by neu-ma-tic** [comment]
maybe try checking the route and redirecting or allowing people to view content based on authentication status in _app.jsx

**Comment by shuding** [comment]
Yes, you can add `rewrites`/`redirects` or using a middleware. Adding things inside _app is another approach. Please search in Next.js docs and related discussion.
